### PR TITLE
add registration of -cache and -origin (SOFTWARE-3368)

### PR DESCRIPTION
**Pull request by zvada**

### CODE REVIEW

File: docs/data/stashcache/install-origin.md [hunk 1]
```diff
@@ -176,7 +176,56 @@ If successful, there should be a file at `/tmp/testfile` with the contents of th
...
 
 
-## Getting help
+Registering the Origin
+----------------------
+To be part of the OSG StashCache Federation, your _origin_ must be
+[registered with the OSG](https://github.com/opensciencegrid/topology#topology).
+To register your resource:
+
+1. Identify the facility, site, and resource group where your _origin_ is hosted.
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]: See previous comment about just pointing them to the general instructions and telling them to use "XRootD origin server" for the service type.

File: docs/data/stashcache/install-origin.md [hunk 2]
```diff
@@ -176,7 +176,56 @@ If successful, there should be a file at `/tmp/testfile` with the contents of th
...
+!!! warning
+    The contact person should be listed in the OSG topology [contacts list](https://topology.opensciencegrid.org/contacts). 
+    If the person or the resource are completely new (e.g. ID and/or GroupID doesn't exist), 
+    you should consider to follow [main OSG registration documentation](/common/registration.md).
+
+1. Using the above information, [create or update](https://github.com/opensciencegrid/topology#how-to-register) the
+   appropriate YAML file, using [this template](https://github.com/opensciencegrid/topology/blob/master/template-resourcegroup.yaml)
+   as a guide.
+
+1. Update information of your resource (longitute/latitude included) in the [stashcp.json](https://github.com/opensciencegrid/StashCache/blob/master/bin/caches.json) by opening pull request.
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]: Origins don't need to be caches.json, right?

File: docs/data/stashcache/install-cache.md [hunk 3]
```diff
@@ -206,3 +206,73 @@ user@host $ condor_status -any -l -const "Name==\"xrootd@`hostname`\""
...
+!!! warning
+    The contact person should be listed in the OSG topology [contacts list](https://topology.opensciencegrid.org/contacts). 
+    If the person or the resource are completely new (e.g. ID and/or GroupID doesn't exist), 
+    you should consider to follow [main OSG registration documentation](/common/registration.md).
+
+1. Using the above information, [create or update](https://github.com/opensciencegrid/topology#how-to-register) the
+   appropriate YAML file, using [this template](https://github.com/opensciencegrid/topology/blob/master/template-resourcegroup.yaml)
+   as a guide.
+
+1. Update information of your resource (longitute/latitude included) in the [stashcp.json](https://github.com/opensciencegrid/StashCache/blob/master/bin/caches.json) by opening pull request.
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply docs/data/stashcache/install-cache.md]: also, it's caches.json, not stashcp.json.

File: docs/data/stashcache/install-origin.md [hunk 4]
```diff
@@ -5,15 +5,15 @@ This document describes how to install a StashCache origin service.  This servic
...
+    The _origin_ must be registered with the OSG prior to joining the data federation. You may start the
+    registration process prior to finishing the installation by [using this link](#registering-the-origin) 
+    along with the basic information like:
 
     * Resource name and hostname.
     * VO associated with this origin server (will be used to determine the origin's namespace prefix).
     * Administrative and security contact.
 
-    Follow the [registration documentation](/common/registration.md) for more information.
+    Detailed walkthrough of the process is described [here](#registering-the-origin).
```
Comment (matyasselmeci) [inline docs/data/stashcache/install-origin.md]: You've got two links to the same place a few lines apart; I don't think you need both.

File: docs/data/stashcache/install-cache.md [hunk 5]
```diff
@@ -206,3 +206,73 @@ user@host $ condor_status -any -l -const "Name==\"xrootd@`hostname`\""
...
+   In order to identify GeoIP information you can use the following tool provided by [MAXMIND](https://www.maxmind.com/en/geoip-demo).  
+
+1. In order to get your cache available to the CVMFS _osg-config_ repository (OASIS), open pull request by adding your resource under `CVMFS_EXTERNAL_URL` [here](https://github.com/opensciencegrid/oasis-server/blob/6aa1492b44ed7d74f1b737f1ea92ace31190e6a2/goc/config-osg/etc/cvmfs/domain.d/osgstorage.org.conf#L7).
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply docs/data/stashcache/install-cache.md]: Honestly though, these files look complicated enough that I don't think users should be modifying them directly. How about telling them to make a ticket, and providing instructions for us on how to help them on the operations pages: http://opensciencegrid.org/operations/.

[the remark this answers is not in the document]
Comment (bbockelm) [reply docs/data/stashcache/install-cache.md]: I agree on this.  For the tricky pieces that @matyasselmeci is referencing, we can just have an internal doc under the "Services" dropdown in the operations docs.

**Comment by bbockelm** [comment]
@zvada, @matyasselmeci  - @djw8605 and I are working on StashCache issues today and are planning to push the requested fixes @matyasselmeci has requested to this branch.

(Marian is actually out today so there shouldn't be any clobbering of changes... just wanted to record that we're starting to work on this right now.)

**Comment by djw8605** [comment]
I am splitting this document.  The stuff about putting the cache in caches.json and putting it in the oasis config, I am moving to ops documentation.  There will be a handoff where those steps are done by the ops team.

File: docs/data/stashcache/install-cache.md [hunk 6]
```diff
@@ -6,14 +6,14 @@ network to cache data frequently used on the OSG, reducing data transfer over th
 decreasing access latency.
 
 !!! note
-    The cache service must be registered with the OSG if it is to be used by clients.  You may start the
-    registration procedure prior to finishing the installation by contacting <support@opensciencegrid.org>
-    with the following details:
+    The _cache_ service must be registered with the OSG if it is to be used by clients. You may start 
+    registration procedure prior to finishing the installation by [using this link](#registering-the-cache) 
+    along with the basic information like:
```
Comment (bbockelm) [inline docs/data/stashcache/install-cache.md]: Why are these details needed if you follow the link?

**Comment by bbockelm** [comment]
@djw8605 @matyasselmeci - can we drive this one home today?  Marian is still out, but I'd like to check this off my list.

**Comment by bbockelm** [comment]
@zvada @djw8605 - please don't forget this document.

**Comment by zvada** [comment]
I'm a bit lost for all the changes after original pull request proposed from `zvada:register-cache`. Should we simply merge it as is now and focus on further edits from there? Or another trick I get the actual version to be worked on?

**Comment by djw8605** [comment]
The changes are so small now since we broke some of this documentation out into the ops pages that the previous reviews are no longer relevant.  @matyasselmeci re-review please.

**Pull request merged**